stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -87,10 +87,195 @@ public enum _ExecutionMode : Equatable {
   }
 }
 
+// TraceContext contains the state as needed to build a trace graph function
+// (TF_Function). As eager ops are executed in tracing mode, their corresponding
+// nodes are added to the trace graph (via `addEagerOpToGraph()`). When the
+// trace is fully formed (via `finalizeAndExecute()`), the trace graph function
+// is run by the eager runtime.
+private class TraceContext {
+  let status: CTFStatus = TF_NewStatus()
+
+  // The trace graph, which will be converted to a trace graph function
+  // (TF_Function) upon finalizing.
+  let graph = TF_NewGraph()
+
+  // The list of inputs to the trace graph function.
+  //
+  // These symbolic tensors corresond to PlaceHolder nodes in the trace graph,
+  // and will be filled in when we execute the trace graph function.
+  //
+  // TODO: If some tensors in `x` are not used within `foo()`, they can be
+  // pruned away in the inputs to the trace graph function.
+  var symbolicInputs: [TF_Output] = []
```

Comment:
nit: doesn't matter for semantics, but shoudl this be TF_Input?

Reply:
Using TF_Output is the convention (e.g. also in TFLowerGraph.cpp). TF_Input appears to be deprecated.